packages/@lwc/template-compiler/src/__tests__/parser.spec.ts

```diff
@@ -475,6 +503,20 @@ describe('expression', () => {
         expect(root.children[0].attrs.title.value).toBe('{myValue}');
     });
 
+    it('escaped brace characters', () => {
+        const { root } = parseTemplate(
+            `<template><select title="&#x007B;x&#x007D;"></select></template>`
```

Comment:
This is the part that I am not confident about. `<select title="&#x007B;x&#x007D;"></select>` is equivalent to `<select title="{x}"></select>` in this format we throw because it is an invalid expression. IMO, equivalent templates should produce the same output. If you want to make it valid, you can put a `\` in front of it.